docs/proposals-accepted/202107-protobuf-openapi-httpapi.md

```diff
@@ -0,0 +1,73 @@
+| type     | title                                                        | status   | menu               |
+| -------- | ------------------------------------------------------------ | -------- | ------------------ |
+| proposal | Expose Thanos APIs to OpenAPI/protobuf and expose on website and UIs | accepted | proposals-accepted |
+
+
+
+## **Expose Thanos APIs to OpenAPI/protobuf and expose on website and UIs**
+
+- Owners:
+  - [@Hangzhi](https://github.com/Hangzhi)
+- Related Tickets:
+  - Program Proposal: https://github.com/thanos-io/thanos/issues/4102
+  - Need for auto-documentation in Prometheus:
+    - https://github.com/prometheus/prometheus/issues/7192
+    - https://github.com/prometheus/prometheus/issues/5567
+- Other docs:
+  - A proposal of LFX mentorship: https://docs.google.com/document/d/1GeyFCk4QkVSrx4pQ4OSmOA1KE_FvmJRi_aq9hr2GoZQ/edit?usp=sharing
+
+This design doc is proposing a design for Thanos HTTP API defined in protobuf and OpenAPI.
+
+## **Motivations**
+
+In order to improve Thanos usage for users, we would like to define our HTTP APIs in protobuf/OpenAPI and expose those in the website. This would allow users to use tools for documentation, validation, type checking, and even code generation to use our APIs efficiently.
+
+### **Pitfalls of the current solution**
+
+- Documentation needs to be written manually.

```

Comment:
```suggestion
- Documentation, Server code and client code boilerplate needs to be written manually.
- It's hard to discover the current API programmatically.
- It's easy to make an accidental breaking changes when modifying API
```

Reply:
I would reword the final point to be something like
> When modifying the API, it is very difficult to know if this breaks downstream users

Or similar :)